applications/RANSApplication/custom_elements/incompressible_potential_flow/incompressible_potential_flow_pressure_element.h

```diff
@@ -0,0 +1,382 @@
+     */
+    IncompressiblePotentialFlowPressureElement(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry)
+    : BaseType(NewId, pGeometry)
+    {
+    }
+
+    /**
+     * Constructor using Properties
+     */
+    IncompressiblePotentialFlowPressureElement(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry,
+        PropertiesType::Pointer pProperties)
+    : BaseType(NewId, pGeometry, pProperties)
+    {
+    }
+
+    /**
+     * Copy Constructor
+     */
+    IncompressiblePotentialFlowPressureElement(
+        IncompressiblePotentialFlowPressureElement const& rOther)
+    : BaseType(rOther)
+    {
+    }
+
+    /**
+     * Destructor
+     */
+    ~IncompressiblePotentialFlowPressureElement() override = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * ELEMENTS inherited from this class have to implement next
+     * Create and Clone methods: MANDATORY
+     */
+
+    /**
+     * creates a new element pointer
+     * @param NewId: the ID of the new element
+     * @param ThisNodes: the nodes of the new element
+     * @param pProperties: the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties) const override
+    {
+        KRATOS_TRY
+        return Kratos::make_intrusive<IncompressiblePotentialFlowPressureElement>(
+            NewId, Element::GetGeometry().Create(ThisNodes), pProperties);
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * creates a new element pointer
+     * @param NewId: the ID of the new element
+     * @param pGeom: the geometry to be employed
+     * @param pProperties: the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        PropertiesType::Pointer pProperties) const override
+    {
+        KRATOS_TRY
+        return Kratos::make_intrusive<IncompressiblePotentialFlowPressureElement>(
+            NewId, pGeom, pProperties);
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * creates a new element pointer and clones the previous element data
+     * @param NewId: the ID of the new element
+     * @param ThisNodes: the nodes of the new element
+     * @param pProperties: the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Clone(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes) const override
+    {
+        KRATOS_TRY
+        return Kratos::make_intrusive<IncompressiblePotentialFlowPressureElement>(
+            NewId, Element::GetGeometry().Create(ThisNodes), Element::pGetProperties());
+        KRATOS_CATCH("");
+    }
+
+    const Variable<double>& GetVariable() const override
+    {
+        return PRESSURE_POTENTIAL;
```

Comment:
Can you specify why is this `PRESSURE_POTENTIAL` used for? Shouldn't the velocity field be post-processed with the `VELOCITY_POTENTIAL` stream function?

Reply:
Yes, It is post processed. But I am using slightly different formulation here, thats why I need PRESSURE_POTENTIAL variable.